src/ucp/rndv/rndv.h

```diff
@@ -12,43 +12,65 @@
 #include <ucs/datastruct/ptr_map.h>
 
 
-enum ucp_rndv_rts_flags {
-    UCP_RNDV_RTS_FLAG_TAG = UCS_BIT(0),
-    UCP_RNDV_RTS_FLAG_AM  = UCS_BIT(1)
-};
+typedef enum {
+    UCP_RNDV_RTS_FLAG_TAG             = UCS_BIT(0),
+    UCP_RNDV_RTS_FLAG_AM              = UCS_BIT(1)
+} UCS_S_PACKED ucp_rndv_rts_flags_t;
 
 
 /*
  * Rendezvous RTS
  */
 typedef struct {
-    ucp_request_hdr_t         sreq;     /* send request on the rndv initiator side */
-    uint64_t                  address;  /* holds the address of the data buffer on the sender's side */
-    size_t                    size;     /* size of the data for sending */
-    uint16_t                  flags;    /* rndv proto flags, as defined by
-                                           ucp_rndv_rts_flags */
+    /* Protocol-specific header */
+    uint64_t          hdr;
+    /* Send request on the rndv initiator side */
+    ucp_request_hdr_t sreq;
+    /* Holds the address of the data buffer on the sender's side */
+    uint64_t          address;
+    /* Size of the data for sending */
+    size_t            size;
+    union {
+        /* RNDV proto flags, as defined by ucp_rndv_rts_flags_t */
+        uint8_t       flags;
+        /* RNDV operation status, used only by old version for TAG protocol,
+         * kept here for wire compatibility
+         */
+        ucs_status_t  status;
+    };
+    /*
+     * 1. Packed rkeys follow
+     * 2. AM only: User header follows, if am->header_length is not 0
+     */
 } UCS_S_PACKED ucp_rndv_rts_hdr_t;
 
 
 /*
  * Rendezvous RTR
  */
 typedef struct {
-    uint64_t                  sreq_id;  /* request ID on the rndv initiator side - sender */
-    uint64_t                  rreq_id;  /* request ID on the rndv receiver side */
-    uint64_t                  address;  /* holds the address of the data buffer on the receiver's side */
-    size_t                    size;     /* size of the data to receive */
-    size_t                    offset;   /* offset of the data in the recv buffer */
-    /* packed rkeys follow */
+    /* Request ID on the rndv initiator side - sender */
+    uint64_t sreq_id;
+    /* Request ID on the rndv receiver side */
+    uint64_t rreq_id;
+    /* Holds the address of the data buffer on the receiver's side */
+    uint64_t address;
+    /* Size of the data to receive */
+    size_t   size;
+    /* Offset of the data in the recv buffer */
+    size_t   offset;
+    /* Packed rkeys follow */
```

Comment:
minor: no need to change, it is not affected by this PR

Reply:
reverted